src/182B30.c

```diff
@@ -2,13 +2,96 @@
 #include "sprite.h"
 #include "effects.h"
 #include "battle/battle.h"
+#include "sprite/npc/BattleWatt.h"
+
+enum StandardPalettes {
+    STANDARD_PAL_POISON     = 1,
+    STANDARD_PAL_DIZZY      = 2,
+    STANDARD_PAL_STATIC     = 3,
+};
+
+#define UNPACK_PAL_R(color) (((color) >> 11) & 0x1F) 
+#define UNPACK_PAL_G(color) (((color) >> 6) & 0x1F) 
+#define UNPACK_PAL_B(color) (((color) >> 1) & 0x1F) 
+#define UNPACK_PAL_A(color) ((color) & 1)
+
+#define PACK_PAL_RGBA(r, g, b, a) (((r) << 11) | ((g) << 6) | ((b) << 1) | (a));
+
+// lerp from A to B as alpha does from 0 to 255
+#define LERP_COMPONENT(a, b, alpha) ((a) * (255 - (alpha)) + (b) * (alpha)) / 255;
+
+enum PalSwapState {
+    PAL_SWAP_HOLD_A     = 0,
+    PAL_SWAP_A_TO_B     = 1,
+    PAL_SWAP_HOLD_B     = 2,
+    PAL_SWAP_B_TO_A     = 3,
+};
+
+// animated palettes for electrified sprites
+// each pair gives { mode, duration }
+u8 StaticPalettesAnim[] = {
+    0, 32,
+    1, 4,
+    2, 2,
+    0, 16,
+    1, 2,
+    0, 64,
+    1, 2,
+    2, 2,
+    0, 28,
+    1, 2,
+    0, 18,
+    1, 4,
+    0, 16,
+    1, 2,
+    0, 80,
+    1, 2,
+    0, 16,
+    2, 2,
+    1, 2,
+    0, 32,
+    1, 2,
+    0, 14,
+    1, 2,
+    2, 2,
+    255
+};
+
+s16 FearPaletteAnimXOffsets[] = { -2, 2, 0, 0, -2, 2, 0, 0, 0, 0, -2, 2, 0, 0, 0, 0, 0, 0, 255 };
+
+s16 ParalyzePaletteAnimXOffsets[] = { -2, 2, 0, 0, -2, 2, 0, 0, 0, 0, -2, 2, 0, 0, 0, 0, 0, 0, 255 };
+
+// animated palettes for Watt
+// each pair gives { mode, duration }
+u8 WattPalettesAnim[] = {
+    1,  2,
```

Comment:
These numbers are right-aligned, while the other arrays in this file are left-aligned.

Also, not sure what `clang-format` is gonna do to this. Maybe something like this could work?
```
#define TUPLE(foo, bar) (foo), (bar)

u8 StaticPalettesAnim[] = {
    TUPLE(0, 32),
    TUPLE(1, 4),
    TUPLE(2, 2),
    255
};
```

Reply:
I think the formatter should be disregarded in these cases. There are many instances of tables like this throughout the code.